src/cr/cube/crunch_cube.py

```diff
@@ -386,25 +380,24 @@ def row_direction_axis(self):
             return 2
         return 1
 
-    def inserted_dim_inds(self, transforms, dim):
-        if not transforms:
+    def _inserted_dim_inds(self, transform_dims, axis):
+        if not transform_dims:
             return []
         inserted_inds = self.inserted_hs_indices()
-        if dim == 0:  # In case of row
-            dim_ind = 0 if self.ndim < 3 else 1
-        elif dim == 1:
-            dim_ind = 1 if self.ndim < 3 else 2
-        return np.array(
-            inserted_inds[dim_ind] if len(inserted_inds) else []
-        )
+        dim_ind = axis if self.ndim < 3 else axis + 1
+        return np.array(inserted_inds[dim_ind] if len(inserted_inds) else [])
 
     @staticmethod
-    def _margin_pruned_indices(margin, insertions):
-        ind_inserted = np.zeros(margin.shape, dtype=bool)
-        if insertions is not None and any(insertions):
-            ind_inserted[insertions] = True
+    def _margin_pruned_indices(margin, inserted_ind, axis):
         pruned_ind = np.logical_or(margin == 0, np.isnan(margin))
-        pruned_ind = np.logical_and(pruned_ind, ~ind_inserted)
+
+        if np.any(margin) and inserted_ind is not None and any(inserted_ind):
```

Comment:
It would be more elegant if you just returned `pruned_ind` on the inverted condition here, and had the rest of the code outdented. At least at first :eyes:

Reply:
I feel like it nicely outlines that the `pruned_ind` is basically starting with the same matrix but if there are insertions we are masking these inserted indices